### PR TITLE
Add membership screening

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Adds discord/discord-api-docs#2396
- Add models
- Add rest requests

Related to #121, #216

### CODE REVIEW

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 1]
```diff
@@ -557,4 +574,28 @@ data class DiscordWelcomeScreen(
...
+    @Serializable
+    data class FormField(
+        @SerialName("field_type")
+        val type: Type,
+        val label: String,
+        val values: Optional<List<String>> = Optional.Missing(),
+        val required: Boolean
+    ) {
+        @Serializable
+        enum class Type {
```
Comment (HopeBaron) [inline common/src/main/kotlin/entity/DiscordGuild.kt:596]: Use a more expresive name than `Type`, maybe `FormFieldType` is a better. name
Also use objects so we can have `Unknown` type in there

**Review by HopeBaron — CHANGES_REQUESTED** [review]
There is no core representation for member screening

**Comment by DRSchlaubi** [comment]
actually, they removed the docs for membership screening so I stopped implementing them until they do a PR adding them again

**Comment by HopeBaron** [comment]
Any updates on this?

**Comment by DRSchlaubi** [comment]
https://discord.com/developers/docs/resources/guild#membership-screening-object still has the warning and I don't see any PR changeing it so I guess no updates

**Comment by HopeBaron** [comment]
Understood :+1:

**Pull request closed without merging**